doc/api/stream.md

```diff
@@ -1011,94 +1010,92 @@ However, this would be better implemented as a [Transform][] stream. See
 // alternative example below under the Transform section.
 
 const Readable = require('stream').Readable;
-const util = require('util');
 
-util.inherits(SimpleProtocol, Readable);
+class SimpleProtocol extends Readable {
+  constructor(source, options) {
+    super(options);
+    if (!(this instanceof SimpleProtocol))
+      return new SimpleProtocol(source, options);
```

Comment:
Not sure how to proceed here. They come from the original example and enforce the singleton, I guessed? Personally I think we can drop them for simplicity anyhow. Shall I proceed by removing the checks here?